zipkin/src/main/java/zipkin/storage/InMemorySpanStore.java

```diff
@@ -69,18 +103,91 @@ public InMemorySpanStore() {
           traceIdTimeStamps.add(traceIdTimeStamp);
           traceIdToSpans.put(span.traceId, span);
           acceptedSpanCount++;
+          retainedSpanCount++;
+          metrics.setTracesRetained(traceIdToSpans.keySet().size());
+          metrics.incrementSpansRetained(1);
 
           for (String serviceName : span.serviceNames()) {
             serviceToTraceIdTimeStamp.put(serviceName, traceIdTimeStamp);
             serviceToSpanNames.put(serviceName, spanName);
           }
         }
       }
+      purge(maxSpans);
     }
 
     @Override public String toString() {
       return "InMemorySpanConsumer";
     }
+
+    /*
+     * Accepts number of spans to remain after purging.  Will delete a trace and all
+     * its spans until this goal is achieved.
+     *
+     * Returns number of spans purged.
```

Comment:
since the description is written like this, maybe `@return` is better that a newline and the same.